### DISCUSSION TITLE
Issue with React Query Not Re-rendering on Successful MSW Response (testing)

### DISCUSSION THREAD

**Discussion by tam-eoh** [TanStack/query#6455] [Q&A] [answered]
Hello everyone,

I'm currently facing an issue with React Query and MSW in my project and would appreciate any insights or suggestions from the community.

Issue Description:

I'm using React Query's useQuery hook for data fetching and MSW to mock the API responses in my tests. However, I've encountered a problem where, even though MSW successfully returns a response, the data fetched by useQuery does not trigger a re-render with the new data.

**Code Example:**

My component
![image](https://github.com/TanStack/query/assets/91381458/c67617cc-b7b7-4adb-81ca-25c540881a00)
![image](https://github.com/TanStack/query/assets/91381458/7cf7e5b5-36cf-4104-b12d-fa25c1bdb4c6)

**Testing**

![image](https://github.com/TanStack/query/assets/91381458/839e4c7a-e12d-4c4c-bdcc-4eddcd0ab73c)
![image](https://github.com/TanStack/query/assets/91381458/88e58400-f197-479f-af84-aa6939f8453e)

**Log info**

![image](https://github.com/TanStack/query/assets/91381458/2c8824cd-9481-44e6-afd5-8f3f43964310)
![image](https://github.com/TanStack/query/assets/91381458/51a72c02-a3d1-4728-bca2-1f09b2f2c6a1)

**Expected Behavior:**

When MSW returns a successful response, useQuery should re-render the component with the new data.

**Environment Details:**

React Query Version: 5.8.7
MSW (Mock Service Worker) Version: 2.0.9
React Test Renderer Version: 18.2.0
Operating System: Ubuntu
React Version: 18.2.0
Testing Approach: Using react-test-renderer for component rendering tests.

**Comment by TkDodo** [comment]
I have a testing setup here where this works just fine:

https://github.com/TkDodo/testing-react-query

**Comment by tam-eoh** [comment]
I have checked your repo, and it works well with MSW. However, in my project, react-query does not trigger a re-render. Could this be due to a version difference in MSW?

Here is my repository: 
https://github.com/tam-eoh/MSW-React-Query. 
Could you please check it again?

**Reply by TkDodo** [reply]
I suggest you try to upgrade my reproduction to the latest msw, and if that fails, I'll take a look. If it works, you can compare yours to mine to see where yours went wrong.

**Reply by tam-eoh** [reply, marked as answer]
I found that the reason it does not get data from useQuery is due to using a different testing library. When using react-test-renderer, it is necessary to update the component in order to trigger a re-render and thus receive the data.

![image](https://github.com/TanStack/query/assets/91381458/968214a6-a754-4c2d-9084-fe27dc860b27)
